src/main/java/org/opensrp/web/config/security/filter/XssPreventionRequestWrapper.java

```diff
@@ -0,0 +1,130 @@
+package org.opensrp.web.config.security.filter;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.JsonNodeType;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.apache.commons.io.IOUtils;
+import org.owasp.encoder.Encode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.servlet.ServletInputStream;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.util.Iterator;
+import java.util.Map;
+
+public class XssPreventionRequestWrapper extends HttpServletRequestWrapper {
+
+	private static Logger logger = LoggerFactory.getLogger(XssPreventionRequestWrapper.class);
+
+	private static ObjectMapper mapper = new ObjectMapper();
+
+	private byte[] rawData;
+
+	private HttpServletRequest request;
+
+	private ResettableServletInputStream servletStream;
+
+	public XssPreventionRequestWrapper(HttpServletRequest request) {
+		super(request);
+		this.request = request;
+		this.servletStream = new ResettableServletInputStream();
+	}
+
+	public void resetInputStream() throws IOException {
+		if (rawData == null) {
+			rawData = IOUtils.toByteArray(this.request.getReader());
+			servletStream.stream = new ByteArrayInputStream(rawData);
+		}
+		servletStream.stream = new ByteArrayInputStream(rawData);
+	}
```

Comment:
This method, is not required since `getInputStream` and `getReader` have been overriden